backend/app/api/routes/dde.py

```diff
@@ -201,27 +202,45 @@ def delete_document_data_example(*, session: SessionDep, current_user: CurrentUs
     session.commit()
     return Message(message="DocumentDataExample deleted successfully")
 
-@router.post("/extract/{name}", response_model=ChatCompletionResponse)
+@router.post("/extract/{name}")
 async def extract_from_file(*, session: SessionDep, current_user: CurrentUser, name: str, upload: UploadFile) -> JSONResponse:
     document_data_extractor = crud.get_document_data_extractor(session=session, name=name)
     if not document_data_extractor:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="DocumentDataExtractor not found")
     if not document_data_extractor.owner_id:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="DocumentDataExtractor has no owner")
     # Build examples
-    examples = tup(*(tup(as_text(document_data_extractor.owner, example.document_id), example.data) for example in document_data_extractor.document_data_examples))
+    examples = tup(*(tup(as_text(document_data_extractor.owner, example.document_id, example.start_page, example.end_page), example.data) for example in document_data_extractor.document_data_examples))
     # Pull data from the file
     if upload.content_type != 'application/pdf':
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="This endpoint can only process pdfs")
     prompt = pdf_reader.as_text(upload.file.read())
-    # TODO Marta can improve this
-    chat_completion_request = ChatCompletionRequest(
-        model='gpt-3.5-turbo',
-        messages=[ChatCompletionMessage(role="system", content=document_data_extractor.prompt)]+
-            [ChatCompletionMessage(role=role, content=content) for example in await examples.evaluate() for role, content in [("user", example[0]), ("assistant", example[1])]]+
-            [ChatCompletionMessage(role="user", content=prompt)]
-    ).model_dump(exclude_unset=True)
-    print(chat_completion_request)
+    if prompt == "":
```

Comment:
can you factor this in a function that is used for prompt,input_text and output_text?